fix: addWatchFile doesn't work if base is specified (fixes #19792)

anatawa12 opened the pull request:
### Description

Fixes #19792 

URLs used in `updateModuleInfo` are relative to base url but normalizeUrl will prepend base url unconditionally.
This PR changes to not prepend base url if it's for pluginImports

<!-- What is this PR solving? Write a clear description or reference the issues it solves (e.g. `fixes #123`). What other alternatives have you explored? Are there any parts you think require more attention from reviewers? -->

<!----------------------------------------------------------------------
Before creating the pull request, please make sure you do the following:

- Read the Contributing Guidelines at https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md.
- Check that there isn't already a PR that solves the problem the same way. If you find a duplicate, please help us reviewing it.
- Update the corresponding documentation if needed.
- Include relevant tests that fail without this PR but pass with it.

Thank you for contributing to Vite!
----------------------------------------------------------------------->

anatawa12 committed:
fix: addWatchFile doesn't work if base is specified

hi-ogawa reviewed (commented):
Thanks for the fix! Not sure which is better, but looking at other code, applying `stripBase` afterward might be more consistent.

Can you also add a test case in `playground/transform-plugin`? I think you can add `playground/transform-plugin/vite.config-base.js` and `playground/transform-plugin/__tests__/base/transform-plugin.spec.ts`. This is the convention to reuse the same test but with slightly different config (you can check other playground like `playground/css/vite.config-xxx`).

anatawa12 committed:
chore: stripBase afterwards instead of skipping preceding base

anatawa12 committed:
test: add test for HMR with a base path

anatawa12 commented:
I updated the implementation and added tests.

Similar to css family I moved test implementation to `tests.ts` and call it in `transform-plugin.spec.ts` and `base/transform-plugin.spec.ts`

sapphi-red commented:
/ecosystem-ci run

sapphi-red reviewed (approved):
Thanks!
